cln_lspd.go

```diff
@@ -0,0 +1,319 @@
+		return nil, 0, err
+	}
+	return FundingTxId, uint32(0), err
+}
+
+func clnIsConnected(clientcln *glightning.Lightning, destination string) error {
+	pubKey := destination
+	peers, err := clientcln.ListPeers()
+	if err != nil {
+		return fmt.Errorf("clientcln.ListPeers() error: %v", err)
+	}
+	for _, peer := range peers {
+		if pubKey == peer.Id {
+			log.Printf("destination online: %v", destination)
+			return nil
+		}
+	}
+	log.Printf("destination offline: %v", destination)
+	return fmt.Errorf("destination offline")
+}
+func OnHtlcAccepted(event *glightning.HtlcAcceptedEvent) (*glightning.HtlcAcceptedResponse, error) {
+	log.Printf("htlc_accepted called\n")
+	onion := event.Onion
+
+	log.Printf("htlc: %v\nchanID: %v\nincoming amount: %v\noutgoing amount: %v\nincoming expiry: %v\noutgoing expiry: %v\npaymentHash: %v\nonionBlob: %v\n\n",
+		event.Htlc,
+		onion.ShortChannelId,
+		event.Htlc.AmountMilliSatoshi, //with fees
+		onion.ForwardAmount,
+		event.Htlc.CltvExpiryRelative,
+		event.Htlc.CltvExpiry,
+		event.Htlc.PaymentHash,
+		onion,
+	)
+
+	paymentHashBytes, err := hex.DecodeString(event.Htlc.PaymentHash)
+	if err != nil {
+		log.Printf("hex.DecodeString(%v) error: %v", event.Htlc.PaymentHash, err)
+		return event.Continue(), fmt.Errorf("hex.DecodeString(%v) error: %w", event.Htlc.PaymentHash, err)
+	}
+
+	paymentHash, paymentSecret, destination, incomingAmountMsat, outgoingAmountMsat, fundingTxID, fundingTxOutnum, err := paymentInfo(paymentHashBytes)
+	if err != nil {
+		log.Printf("paymentInfo(%v)\nfundingTxOutnum: %v\n error: %v", event.Htlc.PaymentHash, fundingTxOutnum, err)
+	}
+
+	if paymentSecret != nil {
+		if fundingTxID == nil {
+			if hex.EncodeToString(paymentHash) == event.Htlc.PaymentHash {
+
+				fundingTxID, fundingTxOutnum, err = clnOpenChannel(clientcln, hex.EncodeToString(paymentHash), hex.EncodeToString(destination), incomingAmountMsat)
+				log.Printf("openclnOpenChannelChannel(%v, %v) err: %v", destination, incomingAmountMsat, err)
+				if err != nil {
+					log.Printf(" clnOpenChannel error: %v", err)
+				}
+
+			} else { //probing
+				failureCode := "TEMPORARY_CHANNEL_FAILURE"
+				if err = clnIsConnected(clientcln, hex.EncodeToString(destination)); err == nil {
+					failureCode = "INCORRECT_OR_UNKNOWN_PAYMENT_DETAILS"
+				}
+				log.Printf(" failureCode %v , Error %v", failureCode, err)
+				return event.Continue(), err
+			}
+
+		}
+		var h chainhash.Hash
+		err = h.SetBytes(fundingTxID)
+		if err != nil {
+			log.Printf("h.SetBytes(%v) error: %v", fundingTxID, err)
+		}
+		channelPoint := wire.NewOutPoint(&h, fundingTxOutnum).String()
+		return clnResumeOrCancel(clientcln, hex.EncodeToString(destination), hex.EncodeToString(fundingTxID), hex.EncodeToString(paymentHash), uint64(outgoingAmountMsat), 99, event, channelPoint)
+
+	}
+
+	return event.Continue(), err
+}
+
+func clnGetChannel(clientcln *glightning.Lightning, destination string, fundingTxID string) string {
```

Comment:
change prototype to return string,bool so we will be able to know if this function found a channel